src/main/java/org/primefaces/renderkit/SelectOneRenderer.java

```diff
@@ -71,5 +77,48 @@ protected Object getSubmittedValues(UIComponent component) {
         return null;
     }
 
+    /**
+     * Recursive method used to find a SelectItem by its value or label.
+     * @param fc FacesContext
+     * @param component the current UI component to find value for
+     * @param converter the converter for the select items
+     * @param selectItems the List of SelectItems
+     * @param valueOrLabel the input value/label to search for
+     * @param byValue true if searching by value false if by label
+     * @return either the SelectItem found or NULL if not found
+     */
+    protected SelectItem findSelectItem(FacesContext fc, UIComponent component, Converter converter, List<SelectItem> selectItems,
+                String valueOrLabel, boolean byValue) {
+        SelectItem foundValue = null;
+        for (int i = 0; i < selectItems.size(); i++) {
+            SelectItem item = selectItems.get(i);
+            if (item instanceof SelectItemGroup) {
+                SelectItemGroup selectItemGroup = (SelectItemGroup) item;
+                if (selectItemGroup.getSelectItems() == null) {
+                    continue;
+                }
+                foundValue = findSelectItem(fc, component,  converter, Arrays.asList(selectItemGroup.getSelectItems()), valueOrLabel, byValue);
```

Comment:
i like it more to do a short "return" here
we can remove a 1 if, 2 breaks, 1 var with that

Reply:
Hmmm can you elaborate?  I am not quite sure I follow.

Reply:
"return findSelectItem(...."

Reply:
Got it. Yeah i only want to break the loop if its found.